spec.bs

```diff
@@ -18,3 +18,50 @@ Default Biblio Status: current
 Markup Shorthands: markdown yes
 Assume Explicit For: yes
 </pre>
+
+
+<h2 id="advertisements">Advertisements</h2>
+
+An <dfn export>advertisement</dfn>
+is a [=struct=] with the following [=struct/items=]:
+
+* An <dfn for=advertisement>render URL</dfn>. The render URL must have “https” as the scheme.
+* An optional <dfn for=advertisement>metadata</dfn> which is text containing opaque JSON encoded data, passed as an object to auction worklet.
+
+
+<h2 id="advertisement-interest-groups">Advertisement Interest Groups</h2>
+
+An <dfn export>advertisement interest group</dfn>
+is a [=struct=] with the following [=struct/items=]:
+ 
+* An <dfn for="advertisement interest group">expiration</dfn> date.   base::Time expiry;
+
+* An <dfn for="advertisement interest group">owner origin</dfn>, an [=origin=] url.
+
+* A <dfn for="advertisement interest group">name</dfn>, a [=string=].
+ 
+* A <dfn for="advertisement interest group">priority</dfn>, a double, initiallized to 0.
+ 
+* An <dfn for="advertisement interest group">enable_bidding_signals_prioritization</dfn>, a boolean, initialized to false.
+ 
+* An optional <dfn for="advertisement interest group">priority_vector</dfn>, A flat [=map=] `<base::flat_map<std::string, double>>`
+ 
+* An optional <dfn for="advertisement interest group">priority_signals_overrides</dfn>, A flat [=map=] `<base::flat_map<std::string, double>>`
+ 
+* An <dfn for="advertisement interest group">execution_mode</dfn>, an ExecutionMode::kCompatibilityMode;

```

Comment:
ExecutionMode::kCompatibilityMode is a Chromium enumeration, we should probably turn this into some kind of spec English.

Reply:
Yes.  Is there another spec where this is defined that we can reference?

Reply:
Found the idl for WorkletExecutionMode, and linked to it.

Reply:
Are you sure we should be using web idl things in an internal data structure?  Specs aren't supposed to reference JS concepts.  Should this just be an integer? and then the JS APIs can convert the enums to integers?

Reply:
I talked to Jeffrey and this sounds fine, we just can't reference WebIDL pieces that are or contain `any`s or `object`s